hive/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableOperations.java

```diff
@@ -0,0 +1,226 @@
+    if (fileIO == null) {
+      fileIO = new HadoopFileIO(conf);
+    }
+
+    return fileIO;
+  }
+
+  @Override
+  protected void doRefresh() {
+    try {
+      org.apache.hadoop.hive.metastore.api.Table hiveTable =
+          metaClients.run(client -> client.getTable(database, tableName));
+
+      Schema schema = LegacyHiveTableUtils.getSchema(hiveTable);
+      PartitionSpec spec = LegacyHiveTableUtils.getPartitionSpec(hiveTable, schema);
+
+      TableMetadata metadata = TableMetadata.newTableMetadata(schema, spec, hiveTable.getSd().getLocation(),
+          LegacyHiveTableUtils.getTableProperties(hiveTable));
+      metadata = metadata.replaceCurrentSnapshot(getDummySnapshot());
+      setCurrentMetadata(metadata);
+    } catch (TException e) {
+      String errMsg = String.format("Failed to get table info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+    setShouldRefresh(false);
+  }
+
+  DirectoryInfo getDirectoryInfo() {
+    Preconditions.checkArgument(current().spec().fields().isEmpty(),
+        "getDirectoryInfo only allowed for unpartitioned tables");
+    try {
+      org.apache.hadoop.hive.metastore.api.Table hiveTable =
+          metaClients.run(client -> client.getTable(database, tableName));
+
+      return LegacyHiveTableUtils.toDirectoryInfo(hiveTable);
+    } catch (TException e) {
+      String errMsg = String.format("Failed to get table info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+  }
+
+  List<DirectoryInfo> getDirectoryInfosByFilter(Expression expression) {
+    Preconditions.checkArgument(!current().spec().fields().isEmpty(),
+        "getDirectoryInfosByFilter only allowed for partitioned tables");
+    try {
+      LOG.info("Fetching partitions for {}.{} with expression: {}", database, tableName, expression);
+      Set<String> partitionColumnNames = current().spec()
+          .identitySourceIds()
+          .stream()
+          .map(id -> current().schema().findColumnName(id))
+          .collect(Collectors.toSet());
+      Expression simplified = HiveExpressions.simplifyPartitionFilter(expression, partitionColumnNames);
+      LOG.info("Simplified expression for {}.{} to {}", database, tableName, simplified);
+      final List<Partition> partitions;
+      // If simplifyPartitionFilter returns TRUE, there are no filters on partition columns or the filter expression is
+      // going to match all partitions
+      if (simplified.op() == Expression.Operation.TRUE) {
+        partitions = metaClients.run(client -> client.listPartitionsByFilter(database, tableName, null, (short) -1));
```

Comment:
TRUE will list all partitions. TRUE does not throw exception in simplification code, it throws exception during conversion to Hive filter string since Hive filter string does not support TRUE literal.